salt/modules/purefb.py

```diff
@@ -0,0 +1,502 @@
+
+- Configure Pure Storage FlashBlade authentication. Use one of the following
+  three methods.
+
+  1) From the minion config
+
+  .. code-block:: yaml
+
+        pure_tags:
+          fb:
+            san_ip: management vip or hostname for the FlashBlade
+            api_token: A valid api token for the FlashBlade being managed
+
+  2) From environment (PUREFB_IP and PUREFB_API)
+  3) From the pillar (PUREFB_IP and PUREFB_API)
+
+:maintainer: Simon Dodsley (simon@purestorage.com)
+:maturity: new
+:requires: purestorage
+:platform: all
+
+.. versionadded:: Oxygen
+
+'''
+
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+import os
+from datetime import datetime
+
+# Import Salt libs
+from salt.ext import six
+from salt.exceptions import CommandExecutionError
+
+# Import 3rd party modules
+try:
+    from purity_fb import PurityFb, FileSystem, FileSystemSnapshot, SnapshotSuffix
+    from purity_fb import rest, NfsRule, ProtocolRule
+    HAS_PURITY_FB = True
+except ImportError:
+    HAS_PURITY_FB = False
+
+__docformat__ = 'restructuredtext en'
+
+VERSION = '1.0.0'
+USER_AGENT_BASE = 'Salt'
+
+__virtualname__ = 'purefb'
+
+
+def __virtual__():
+    '''
+    Determine whether or not to load this module
+    '''
+    if HAS_PURITY_FB:
+        return __virtualname__
+    return (False, 'purefb execution module not loaded: purity_fb python library not available.')
+
+
+def _get_blade():
+    '''
+    Get Pure Storage FlasBlade configuration
+
+    1) From the minion config
+        pure_tags:
+          fb:
+            san_ip: management vip or hostname for the FlashBlade
+            api_token: A valid api token for the FlashBlade being managed
+    2) From environment (PUREFB_IP and PUREFB_API)
+    3) From the pillar (PUREFB_IP and PUREFB_API)
+
+  '''
+
+    try:
+        blade_name = __opts__['pure_tags']['fb'].get('san_ip')
+        api_token = __opts__['pure_tags']['fb'].get('api_token')
+        if blade_name and api:
+            blade = PurityFb(blade_name)
+            blade.disable_verify_ssl()
+    except (KeyError, NameError, TypeError):
+        try:
+            blade_name = os.environ.get('PUREFB_IP')
+            api_token = os.environ.get('PUREFB_API')
+            if blade_name:
+                blade = PurityFb(blade_name)
+                blade.disable_verify_ssl()
+        except (ValueError, KeyError, NameError):
+            try:
+                api_token = __pillar__['PUREFB_API']
+                blade = PurityFb(__pillar__['PUREFB_IP'])
+                blade.disable_verify_ssl()
+            except (KeyError, NameError):
+                raise CommandExecutionError('No Pure Storage FlashBlade credentials found.')
+    try:
+        blade.login(api_token)
+    except Exception:
+        raise CommandExecutionError('Pure Storage FlashBlade authentication failed.')
+    return blade
+
+
+def _get_fs(name, blade):
+    '''Private function to check filesystem'''
```

Comment:
Can you expand this out to all separate lines? I know that's picky, but Salt style is to not have docs on one line like this.